### PR TITLE
Extend support on search param

### LINKED ISSUE #1565: Extend support on search param
### Description

Currently the `search` request param only supports filtering of the results based on the validator `name`.
We need to extend the support to enable filtering of the results based on `address` and `publicKey` as well.

### Acceptance Criteria

- Filtering of results is enabled based on `address` and `publicKey` for the following endpoints
  - `/pos/stakes`
  - `/pos/stakers`
  - `/pos/validators`
- Corresponding integration testcases are added
- Swagger is up-to-date

### PR DESCRIPTION
**Pull request by vardan10**
### What was the problem?
This PR resolves #1565

### How was it solved?

- [x] Filtering of results is enabled based on address and publicKey for /pos/stakes
- [x] Filtering of results is enabled based on address and publicKey for /pos/stakers
- [x] Filtering of results is enabled based on address and publicKey for /pos/validators
- [x] Corresponding integration testcases are added
- [x] Swagger is up-to-date

### How was it tested?

- [x] Running testcases
- [x] Swagger documentation

### CODE REVIEW

File: services/blockchain-indexer/shared/dataService/business/pos/stakers.js [hunk 1]
```diff
@@ -96,10 +96,17 @@ const getStakers = async params => {
 	if (params.search) {
 		const stakerAccountsInfo = await accountsTable.find(
 			{
-				search: {
+				search: [{
 					property: 'name',
 					pattern: params.search,
-				},
+				}],
+				onSearch: [{
```
Comment (nagdahimanshu) [inline services/blockchain-indexer/shared/dataService/business/pos/stakers.js:103]: We don't have `onSearch`. Do you mean with `orSearch`?

Comment (nagdahimanshu) [inline services/blockchain-indexer/shared/dataService/business/pos/stakers.js:103]:
```suggestion
				orSearch: [{
```

File: services/blockchain-indexer/shared/dataService/business/pos/stakes.js [hunk 2]
```diff
@@ -49,10 +49,18 @@ const getStakes = async params => {
...
 	const accountInfoQueryFilter = {};
 	if (params.search) {
-		accountInfoQueryFilter.search = {
+		accountInfoQueryFilter.search = [{
 			property: 'name',
 			pattern: params.search,
-		};
+		}];
+
+		accountInfoQueryFilter.orSearch = [{
```
Comment (nagdahimanshu) [inline services/blockchain-indexer/shared/dataService/business/pos/stakes.js:57]: Same here.

File: services/gateway/apis/http-version3/methods/modules/pos/posValidators.js [hunk 3]
```diff
@@ -28,7 +28,7 @@ module.exports = {
 		publicKey: { optional: true, type: 'string', pattern: regex.PUBLIC_KEY },
 		name: { optional: true, type: 'string', pattern: regex.NAME_CSV, max: 209, altSwaggerKey: 'validatorNameCsv' },
 		status: { optional: true, type: 'string', pattern: regex.POS_VALIDATOR_STATUS, altSwaggerKey: 'validatorStatus' },
-		search: { optional: true, type: 'string', min: 1, altSwaggerKey: 'validatorSearch' },
+		search: { optional: true, type: 'string', min: 1, altSwaggerKey: 'search' },
```
[the remark this answers is not in the document]
Comment (sameersubudhi) [reply services/gateway/apis/http-version3/methods/modules/pos/posValidators.js:31]: Please add the regex as a constant in `services/gateway/shared/regex.js` if not already present and reuse everywhere necessary.

File: services/gateway/shared/regex.js [hunk 4]
```diff
@@ -16,6 +16,7 @@
 const PUBLIC_KEY = /^([A-Fa-f0-9]{2}){32}$/;
 const ADDRESS_LISK32 = /^lsk[a-hjkm-z2-9]{38}$/;
 const ADDRESS_LISK32_CSV = /^(lsk[a-hjkm-z2-9]{38}){1}(,lsk[a-hjkm-z2-9]{38})*$/;
+const PARTIAL_NAME_OR_ADDRESS_OR_PUBLIC_KEY = /^(?:([A-Fa-f0-9]{2}){0,32}|lsk[a-hjkm-z2-9]{0,38}|[\w!@$&.]{3,20})$/;
```
Comment (sameersubudhi) [inline services/gateway/shared/regex.js:19]:
```suggestion
const PARTIAL_SEARCH = ^[\w!@$&.]{1,64}$/; // Supports address, publicKey & name
```

**Review by sameersubudhi — APPROVED** [review]
Looks fine but the regex could be updated as following.

**Review by nagdahimanshu — APPROVED** [review]
LGTM 👍

**Review by sameersubudhi — APPROVED** [review]

**Review by nagdahimanshu — APPROVED** [review]
Looks good 👍

**Pull request merged**